src/Workspaces/Core/Portable/Shared/Extensions/SemanticModelExtensions.cs

```diff
@@ -176,16 +176,34 @@ private static ISymbol MapSymbol(ISymbol symbol, ITypeSymbol type)
             SyntaxToken token,
             CancellationToken cancellationToken)
         {
-            var aliasSymbol = semanticModel.GetAliasInfo(token.Parent, cancellationToken);
+            IAliasSymbol aliasSymbol;
+            ITypeSymbol type;
+            ISymbol declaredSymbol;
+            ImmutableArray<ISymbol> allSymbols;
 
-            var bindableParent = syntaxFacts.GetBindableParent(token);
-            var type = semanticModel.GetTypeInfo(bindableParent, cancellationToken).Type;
-
-            var declaredSymbol = MapSymbol(semanticFacts.GetDeclaredSymbol(semanticModel, token, cancellationToken), type);
-            var allSymbols = semanticModel.GetSymbolInfo(bindableParent, cancellationToken)
-                                          .GetBestOrAllSymbols()
-                                          .WhereAsArray(s => !s.Equals(declaredSymbol))
-                                          .SelectAsArray(s => MapSymbol(s, type));
+            var overriddingIdentifier = syntaxFacts.GetDeclarationIdentifierIfOverride(token);
+            if (overriddingIdentifier.HasValue)
+            {
+                // on an "override" token, we'll find the overridden symbol
+                aliasSymbol = null;
+                var overriddingSymbol = semanticFacts.GetDeclaredSymbol(semanticModel, overriddingIdentifier.Value, cancellationToken);
+                var overriddenSymbol = overriddingSymbol.GetOverriddenMember();
+                type = null;
```

Comment:
It appears that "type" here should be the type _of_ the symbol, which I guess would be the return type of `overriddenSymbol`?

Reply:
Can we add doc comments to the members of this thing so the distinction is clear?

Reply:
I can't speak to what Type is used for, so it's possible it doesn't matter in our scenario. But in that case a comment as to why might be appropriate.

Reply:
Digging further: it looks like Type is used for quick info and go to definition as a fallback for if it's invoked on a numeric literal. So null is fine (because we're providing a symbol in the other places), but a comment to the effect we're doing that intentionally might be good.